data/scripts/dependency-resolver.py

```diff
@@ -80,10 +81,23 @@ def find_makefile_var(self, v):
             return var["value"]
         return ""
 
-def run_command(cmd):
+    def add_log_content(self, line):
+        self.log_content.append(line)
+
+    def write_log_content(self, path):
+        if not self.log_content:
+            return
+
+        with open(path, "w+") as f:
+            for line in self.log_content:
+                f.write("%s\n" % line)
+
```

Comment:
do not cache this to be later written to the file. Write to the file and flush, so if it stops for some reason (ie: exception, user's ^C) we can have parts of the log in the disk.